### PR TITLE
FAT2_156_add_search_by_postcode

### PR DESCRIPTION
**Pull request by robyoung91**

### CODE REVIEW

File: src/SFA.DAS.Location.Application.UnitTests/Search/Queries/WhenHandlingTheSearchLocationRequest.cs [hunk 1]
```diff
@@ -1,33 +1,35 @@
...
+using SFA.DAS.Location.Domain.Models;
 using SFA.DAS.Testing.AutoFixture;
 
-namespace SFA.DAS.Location.Application.UnitTests.Location.Queries
+namespace SFA.DAS.Location.Application.UnitTests.Search.Queries
 {
     public class WhenHandlingTheSearchLocationRequest
     {
-        [Test, MoqAutoData]
+        [Test, MoqInlineAutoData("Man")]
```
Comment (PeterDavidWilliams) [inline src/SFA.DAS.Location.Application.UnitTests/Search/Queries/WhenHandlingTheSearchLocationRequest.cs:18]: Is this test case necessary? Looking at the test, it looks like it _should_ work without setting the search term

**Review by PeterDavidWilliams — CHANGES_REQUESTED** [review]

**Review by PeterDavidWilliams — APPROVED** [review]

File: src/SFA.DAS.Location.Domain/Interfaces/IPostcodeApiService.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+﻿using SFA.DAS.Location.Domain.Models;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.Location.Infrastructure.ApiClient
+{
+    public interface IPostcodeApiService
+    {
+        Task<IEnumerable<SuggestedLocation>> GetAllStartingWithOutcode(string query, int resultCount = 10);
```
Comment (dashton82) [inline src/SFA.DAS.Location.Domain/Interfaces/IPostcodeApiService.cs:9]: Not sure we need a default here for the result count - its defaulted on the controller action as well

File: src/SFA.DAS.Location.Infrastructure.UnitTests/ApiClient/WhenGettingDataFromPostcodeIO.cs [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            var client = new HttpClient(httpMessageHandler.Object);
+            var postcodeService = new PostcodeApiService(client);
+
+            //Act
+            var actual = await postcodeService.GetPostcodeData(query);
+
+            //Assert
+            actual.Should().BeEquivalentTo(postcodeResponse.Result);
+        }
+    }
```
Comment (dashton82) [inline src/SFA.DAS.Location.Infrastructure.UnitTests/ApiClient/WhenGettingDataFromPostcodeIO.cs:82]: Missing a test case for the notfound response and returning null - I'd also split out the two methods under test into separate test files

File: src/SFA.DAS.Location.Domain/Models/SuggestedLocation.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        {
+            return new SuggestedLocation
+            {
+                Lat = source.Lat,
+                Long = source.Long,
+                Postcode = source.Postcode
+            };
+        }
+
+        public static implicit operator SuggestedLocation(Domain.Entities.Location source)
```
Comment (dashton82) [inline src/SFA.DAS.Location.Domain/Models/SuggestedLocation.cs:24]: Doesn't appear to be a test for this mapping

**Review by dashton82 — CHANGES_REQUESTED** [review]

**Review by dashton82 — APPROVED** [review]
